### ISSUE TITLE
Proposal to provide interface for non-zero copy compression/decompression

### ISSUE THREAD

**Issue by xuchuanyin** [luben/zstd-jni#68]
Hi community:
I'm planning to move to ZSTD from Snappy codec, but now we faced some problems in missing corresponding interfaces in Java, so I raised this issue.

Snappy-java provides some zero-copy interfaces for compression and decompression, which is useful when we are using **unsafe** at the same time. So I proposed Zstd-jni to provide these interfaces also.

See the code mentioned above is shown in snippet [Zero-copy interfaces provided by Snappy-Java](https://github.com/xerial/snappy-java/blob/master/src/main/java/org/xerial/snappy/Snappy.java#L379-L407)

**Comment by luben** [comment]
Yes, I can add that. The method signatures will be (srcPtr, srcLen, dstPtr, dstLen) in order to make it safer: not overflow the destination buffer by mistake

**Comment by luben** [comment]
Hi, v1.3.5-3 is on the way to maven central. It adds 2 new methods: compressUnsafe and decompressUnsafe that take the source and destination as longs (unsafe pointers).

Any feedback is welcome.

**Comment by xuchuanyin** [comment]
Thanks, looking forward to that release

**Comment by luben** [comment]
It's already on maven-central, v1.3.5-3. Here is the apidoc for the added methods:

https://static.javadoc.io/com.github.luben/zstd-jni/1.3.5-3/com/github/luben/zstd/Zstd.html#compressUnsafe-long-long-long-long-int-
https://static.javadoc.io/com.github.luben/zstd-jni/1.3.5-3/com/github/luben/zstd/Zstd.html#decompressUnsafe-long-long-long-long-
